src/components/Carousel/FirstVedioCarousel.jsx

```diff
@@ -0,0 +1,49 @@
+import "./vedioCarousel.css";
+import { Carousel } from "./Carousel";
+import { firstVedioCarouselData } from "../../utils/utilityArrays";
+import { useNavigate } from "react-router-dom";
+import { useFilterData } from "../../contexts/filterData-context";
+
+const FirstVedioCarousel = () => {
+
+    const navigate = useNavigate();
+    const { dispatchOfProductsBeingShown } = useFilterData();
+
+    const handleFirstVedioCarouselClick = (eachVedioData) => {
+        dispatchOfProductsBeingShown({ type: "clear" });
+
+        if(eachVedioData.hasOwnProperty('category')){
+            dispatchOfProductsBeingShown({ type: "filterBySection", payload: eachVedioData.category })
+        }else
+            if(eachVedioData.hasOwnProperty('hasDiscount')){
+                dispatchOfProductsBeingShown({ type: "filterByProducts", payload: "hasDiscount" })
+            }else
+                if(eachVedioData.hasOwnProperty('onSale')){
+                    dispatchOfProductsBeingShown({ type: "filterByProducts", payload: "isInSale" })
+                }
+
+        navigate("/Productlisting")
+    }
+
+    return (
+        <div className="ap_first-vedio-carousel" >
+            <Carousel >
+                {
+                    firstVedioCarouselData.map((eachVedioData, index) => {
+                        return (
+                            <div className="ap_first-carousel-vedio-data" key={ index }>
+                                <video src={ eachVedioData.carouselVedio } autoPlay loop muted />
```

Comment:
pass whole object and access key which you want in the component just a suggestion

Reply:
> pass whole object and access key which you want in the component just a suggestion

could you please index a more onto your point here, bcz i have been doing the same here, i m passing the whole object ans accessing the key i want to use!! or specifically which part you are highlighting!!??